src/Features/Core/Portable/ExternalAccess/UnitTesting/API/INewUnitTestingIncrementalAnalyzerProviderImplementation.cs

```diff
@@ -0,0 +1,11 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.UnitTesting.Api
+{
+    internal interface INewUnitTestingIncrementalAnalyzerProviderImplementation
```

Comment:
@shyamnamboodiripad this is an api you would implement/export.

Reply:
Ah is there a change in the interface? Can we not keep the interface unchanged (considering UT does not access the crawler directly)? We currently implement `Microsoft.CodeAnalysis.ExternalAccess.UnitTesting.Api.IUnitTestingIncrementalAnalyzerImplementation` and `Microsoft.CodeAnalysis.ExternalAccess.UnitTesting.Api.IUnitTestingIncrementalAnalyzerProviderImplementation`...

Reply:
No.  We need to make sure we don't run you in both the existing crawler and the new crawler :)  

This way, we can add the new crawler in and merge that in.

Then you can switch to this new API (and remove hte old api), validating that it only lights up in the new crawler.   

Then we can remove the old apis.  

--

(if i'm missing a simpler way to do this, lmk).

Reply:
I see that makes sense. I am guessing the interface shape stays pretty much identical and the only changes involved would be around naming.